### ISSUE TITLE
IPython console doens't work with Pygments 2.0rc1

### ISSUE THREAD

**Issue by spyder-issues-migrator** [spyder-ide/spyder#2042]
_From [ubi...@gmail.com](https://code.google.com/u/110890017321570243418/) on 2014-11-06T16:23:16Z_

Spyder Version:  2.3.1
Python Version:  3.4.2
Qt Version    :  4.8.6, PyQt4 (API v2) 4.11.2 on Linux
pyflakes >=0.6.0:  0.8.1 (OK)
pep8 >=0.6      :  1.4.6 (OK)
IPython >=0.13  :  2.3.0 (OK)
pygments >=1.6  :  2.0rc1 (OK)
pandas >=0.13.1 :  None (NOK)
sphinx >=0.6.6  :  1.2.3 (OK)
psutil >=0.3    :  None (NOK)
rope >=0.9.2    :  None (NOK)
matplotlib >=1.0:  1.3.1 (OK)
sympy >=0.7.0   :  None (NOK)
pylint >=0.25   :  1.3.0 (OK)

uname -a:
Linux 3.14-2-amd64 `#1` SMP Debian 3.14.15-2 (2014-08-09) x86_64 GNU/Linux 

What steps will reproduce the problem?

 1. load spyder with ipython console support
2. type any command: import/from/for/... 3. 

What is the expected output? What do you see instead?

 It happened after last system update.

After step 1-2 (on three computers tested), pops up "internal console" with error:

Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/IPython/qt/console/frontend_widget.py", line 60, in highlightBlock
    super(FrontendHighlighter, self).highlightBlock(string)
  File "/usr/lib/python3/dist-packages/IPython/qt/console/pygments_highlighter.py", line 115, in highlightBlock
    for token, text in self._lexer.get_tokens(string):
  File "/usr/lib/python3/dist-packages/pygments/lexer.py", line 193, in streamer
    for i, t, v in self.get_tokens_unprocessed(text):
  File "/usr/lib/python3/dist-packages/IPython/qt/console/pygments_highlighter.py", line 30, in get_tokens_unprocessed
    for item in action(self, m):
.....
and every next command, error repeated..... 

Please provide any additional information below

.

**Attachment:** [spyder_error.png](http://code.google.com/p/spyderlib/issues/detail?id=2042)

_Original issue: http://code.google.com/p/spyderlib/issues/detail?id=2042_

**Comment by spyder-issues-migrator** [comment]
_From [ccordoba12](https://code.google.com/u/ccordoba12/) on 2014-11-06T16:44:59Z_

This is happening because you're using pygments 2.0rc1. Please use pygments 1.6 until this is fixed by the IPython guys (or by me :-).

This is the upstream issue: https://github.com/ipython/ipython/issues/6877

**Summary:** IPython console doens't work with Pygments 2.0rc1  \<span class="oldvalue"> (was: pops up "internal console" with error) \</span>  
  **Status:** Accepted  
  **Labels:** -OpSys-Linux -Component-PyQt

**Comment by spyder-issues-migrator** [comment]
_From [ccordoba12](https://code.google.com/u/ccordoba12/) on 2014-11-06T16:45:31Z_

**Labels:** MS-v2.3.3

**Comment by spyder-issues-migrator** [comment]
_From [ubi...@gmail.com](https://code.google.com/u/110890017321570243418/) on 2014-11-07T01:48:59Z_

OK, I understand. :) Thnx.

Oh, Debian _testing_, _testing_, _testing_.... :-)

2014-11-07 3:45 GMT+03:00 \<spyderlib@googlecode.com>:

**Comment by spyder-issues-migrator** [comment]
_From [ccordoba12](https://code.google.com/u/ccordoba12/) on 2014-11-07T09:06:33Z_

Don't worry, it help us to work things out with time :-)

By the way, there is already a pull request open to fix this, which will be released in IPython 3.0

**Comment by spyder-issues-migrator** [comment]
_From [ccordoba12](https://code.google.com/u/ccordoba12/) on 2014-11-18T14:16:29Z_

issue #2055 has been merged into this issue.

**Comment by spyder-issues-migrator** [comment]
_From [ccordoba12](https://code.google.com/u/ccordoba12/) on 2014-11-18T14:17:26Z_

**Labels:** -MS-v2.3.3 MS-v2.3.2

**Comment by spyder-issues-migrator** [comment]
_From [ccordoba12](https://code.google.com/u/ccordoba12/) on 2014-11-19T12:23:15Z_

This issue was closed by revision fc906f49c9c1 .

**Status:** Fixed

**Comment by spyder-issues-migrator** [comment]
_From [ubi...@gmail.com](https://code.google.com/u/110890017321570243418/) on 2014-11-20T23:09:59Z_

Oh. That's great!!
Works fine now! :)
Thnx. :)

2014-11-19 23:23 GMT+03:00 \<spyderlib@googlecode.com>:

**Comment by spyder-issues-migrator** [comment]
_From [ccordoba12](https://code.google.com/u/ccordoba12/) on 2014-11-21T10:35:17Z_

Thanks. IPython 3 will take still some months in getting out, so I preferred to avoid further reports about this problem :-)

**Comment by spyder-issues-migrator** [comment]
_From [ccordoba12](https://code.google.com/u/ccordoba12/) on 2014-12-04T13:07:16Z_

issue #2073 has been merged into this issue.
